### PR TITLE
session: add session variable `tidb_optimizer_selectivity_level`.

### PR DESCRIPTION
**Pull request by winoros**
Will take effect in future.
PTAL @coocood @zz-jason @lamxTyler

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -1270,7 +1270,8 @@ const loadCommonGlobalVarsSQL = "select HIGH_PRIORITY * from mysql.global_variab
 	variable.TiDBIndexLookupJoinConcurrency + quoteCommaQuote +
 	variable.TiDBIndexSerialScanConcurrency + quoteCommaQuote +
 	variable.TiDBHashJoinConcurrency + quoteCommaQuote +
-	variable.TiDBDistSQLScanConcurrency + "')"
+	variable.TiDBDistSQLScanConcurrency + quoteCommaQuote +
+	variable.TiDBOptimizerSelectivityLevel + "')"
```
Comment (shenli) [inline session/session.go]: We do not need to make it a global variable.

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -319,6 +319,8 @@ type SessionVars struct {
 	MemQuotaIndexLookupJoin int64
 	// MemQuotaNestedLoopApply defines the memory quota for a nested loop apply executor.
 	MemQuotaNestedLoopApply int64
+	// OptimizationLevel defines the level of tidb's planner.
+	OptimizationLevel int64
```
Comment (lamxTyler) [inline sessionctx/variable/session.go]: Rename it?

File: sessionctx/variable/session.go [hunk 3]
```diff
@@ -319,6 +319,8 @@ type SessionVars struct {
 	MemQuotaIndexLookupJoin int64
 	// MemQuotaNestedLoopApply defines the memory quota for a nested loop apply executor.
 	MemQuotaNestedLoopApply int64
+	// OptimizerSelectivityLevel defines the level of tidb's planner.
```
Comment (lamxTyler) [inline sessionctx/variable/session.go]: The comment needs to be updated as well.

File: sessionctx/variable/varsutil.go [hunk 4]
```diff
@@ -128,7 +128,7 @@ func TiDBOptOn(opt string) bool {
 	return strings.EqualFold(opt, "ON") || opt == "1"
 }
 
-func tidbOptPositiveInt(opt string, defaultVal int) int {
+func tidbOptInt32(opt string, defaultVal int) int {
```
Comment (zz-jason) [inline sessionctx/variable/varsutil.go]: I think the old name is more specific, maybe we should revert this name refactor ?

Comment (winoros) [reply sessionctx/variable/varsutil.go]: it has another function called `tidbOptInt64`

Comment (zz-jason) [reply sessionctx/variable/varsutil.go]: `tidbOptInt64` doesn't check whether this values is positive, while `tidbOptPositiveInt` does.

Comment (zz-jason) [reply sessionctx/variable/varsutil.go]: maybe `tidbOptPositiveInt32` is a good name ?

File: sessionctx/variable/tidb_vars.go [hunk 5]
```diff
@@ -98,6 +98,9 @@ const (
 
 	// tidb_enable_streaming enables TiDB to use streaming API for coprocessor requests.
 	TiDBEnableStreaming = "tidb_enable_streaming"
+
+	// tidb_optimizer_selectivity_level is used for control the content of each operator's statistic information.
```
Comment (zz-jason) [inline sessionctx/variable/tidb_vars.go]: may be we can simplify the comment to "is used to control the selectivity estimation level"?

File: sessionctx/variable/session.go [hunk 6]
```diff
@@ -319,6 +319,9 @@ type SessionVars struct {
 	MemQuotaIndexLookupJoin int64
 	// MemQuotaNestedLoopApply defines the memory quota for a nested loop apply executor.
 	MemQuotaNestedLoopApply int64
+	// OptimizerSelectivityLevel defines the what content operator will hold in its statistic information. The higher it
+	// is, the more complex content operator will hold.
```
Comment (zz-jason) [inline sessionctx/variable/session.go]: ditto

File: sessionctx/variable/varsutil.go [hunk 7]
```diff
@@ -128,7 +128,7 @@ func TiDBOptOn(opt string) bool {
 	return strings.EqualFold(opt, "ON") || opt == "1"
 }
 
-func tidbOptPositiveInt(opt string, defaultVal int) int {
+func tidbOptPositiveInt32(opt string, defaultVal int) int {
```
Comment (coocood) [inline sessionctx/variable/varsutil.go]: Why change the function name?

**Pull request merged**